lib/webResource.ts

```diff
@@ -9,6 +9,22 @@ import { generateUuid } from "./util/utils";
 export type HttpMethods = "GET" | "PUT" | "POST" | "DELETE" | "PATCH" | "HEAD" | "OPTIONS" | "TRACE";
 export type HttpRequestBody = Blob | string | ArrayBuffer | ArrayBufferView | (() => NodeJS.ReadableStream);
 
+/**
+ * Fired in response to upload or download progress (browser only).
+ */
+export type TransferProgressEvent = {
+  /**
+   * The number of bytes loaded so far.
+   */
+  loaded: number,
+
+  /**
+   * The total number of bytes that will be loaded.
+   * If the total number of bytes is unknown, this property will be undefined.
+   */
+  total?: number
```

Comment:
I vote that this be called `totalBytes`. I'm almost always in favor of including a variable's units in its name.